pages/guides/[slug].tsx

```diff
@@ -0,0 +1,43 @@
+import { GetStaticPaths, GetStaticProps } from 'next'
+
+import Article, { Props as ArticleProps } from '../../components/Article'
+import getQueryParameter from '../../util/getQueryParameters'
+import loadAllRecords from '../../util/loadAllRecords'
+import loadCollections from '../../util/loadCollections'
+import loadMarkdownFile from '../../util/loadMarkdownFile'
+import omitUndefinedFields from '../../util/omitUndefinedFields'
+import serializeMdxSource from '../../util/serializeMdxSource'
+import slugToTitleCase from '../../util/slugToTitleCase'
+
+export const getStaticPaths: GetStaticPaths = async () => {
+    const guides = await loadAllRecords('guides', true)
+    const paths = guides.map(guide => ({ params: { slug: guide.slug } }))
+    return {
+        paths,
+        fallback: false,
+    }
+}
+
+export const getStaticProps: GetStaticProps<ArticleProps> = async context => {
+    const slug = getQueryParameter(context.params, 'slug')
+    const baseDirectory = 'guides'
+    const markdownFile = await loadMarkdownFile(baseDirectory, `${slug}.md`)
+    const { serializeResult, toc } = await serializeMdxSource(markdownFile)
+    const recordAuthor = markdownFile.frontMatter.author ? slugToTitleCase(markdownFile.frontMatter.author) : ''
+    return {
+        props: omitUndefinedFields({
+            title: markdownFile.frontMatter.title,
+            alternateTitle: markdownFile.frontMatter.alternateTitle,
+            author: recordAuthor,
+            tags: markdownFile.frontMatter.tags,
+            image: markdownFile.frontMatter.image,
+            socialImage: markdownFile.frontMatter.socialImage,
+            description: markdownFile.frontMatter.description,
+            toc,
+            mdxSource: serializeResult,
+            slug,
+        }),
+    }
+}
+
+export default Article
```

Comment:
Is there an opportunity for us to combine this with `[slug].tsx`? Looks like next.js supports catching all paths, I found this documentation page: https://nextjs.org/docs/routing/dynamic-routes#catch-all-routes

I am not sure if that would go against the thought of compartmentalizing the different content types, just wanted to point out the very similar files and curious if we could have a top-level way to get to all the paths.

Reply:
So I think if we are going to do that, we should probably omit adding another content type for now, since we wouldn't be taking advantage of Next's native routing/namespacing conventions and opportunities. The  top-level `guides` and `posts` directories might be helpful for contributors, but we would be treating these records as equivalent in our `util` methods, and handling any styling customization requests via `location` logic passed to our components, which I think could get messy down the road.

Happy to do whatever you think makes the most sense at this stage! But I think that if we anticipate having more `guides` in the longterm, it would make sense to namespace them at the `pages/<directory>` level.